model/fieldtypes/Varchar.php

```diff
@@ -50,11 +50,14 @@ public function getSize() {
  	 * @see DBField::requireField()
  	 */
 	public function requireField() {
+		$charset = Config::inst()->get('MySQLDatabase', 'charset');
+		$collation = Config::inst()->get('MySQLDatabase', 'collation');
+
 		$parts = array(
 			'datatype'=>'varchar',
 			'precision'=>$this->size,
-			'character set'=>'utf8',
-			'collate'=>'utf8_general_ci',
+			'character set'=> $charset,
+			'collate'=> $collation,
```

Comment:
If that's the case, maybe it's best to drop all of the mysql-specific code from the DBField subclasses. Leave them as is for now, and maybe add a comment to remove these in 4.0. (to prevent semver regressions).